contracts/src/PriceFeeds/FXPriceFeed.sol

```diff
@@ -1,50 +1,144 @@
     bool public isShutdown;
 
-    constructor(address _oracleAdapterAddress, address _rateFeedID, address _borrowerOperationsAddress, address _watchdogAddress) {
+    /// @notice Emitted when the watchdog address is updated
+    /// @param _oldWatchdogAddress The previous watchdog address
+    /// @param _newWatchdogAddress The new watchdog address
+    event WatchdogAddressUpdated(address indexed _oldWatchdogAddress, address indexed _newWatchdogAddress);
+
+    /// @notice Emitted when the contract is shutdown due to oracle failure
+    event FXPriceFeedShutdown();
+
+    /**
+     * @notice Contract constructor
+     * @param disableInitializers Boolean to disable initializers for implementation contract
+     */
+    constructor(bool disableInitializers) {
+      if (disableInitializers) {
+        _disableInitializers();
+      }
+    }
+
+    /**
+    * @notice Initializes the FXPriceFeed contract
+    * @param _oracleAdapterAddress The address of the OracleAdapter contract
+    * @param _rateFeedID The address of the rate feed ID
+    * @param _borrowerOperationsAddress The address of the BorrowerOperations contract
+    * @param _watchdogAddress The address of the watchdog contract
+    * @param _initialOwner The address of the initial owner
+    */
+    function initialize(
+        address _oracleAdapterAddress,
+        address _rateFeedID,
+        address _borrowerOperationsAddress,
+        address _watchdogAddress,
+        address _initialOwner
+    ) external initializer {
+        require(_oracleAdapterAddress != address(0), "FXPriceFeed: ZERO_ADDRESS");
+        require(_rateFeedID != address(0), "FXPriceFeed: ZERO_ADDRESS");
+        require(_borrowerOperationsAddress != address(0), "FXPriceFeed: ZERO_ADDRESS");
+        require(_watchdogAddress != address(0), "FXPriceFeed: ZERO_ADDRESS");
+        require(_initialOwner != address(0), "FXPriceFeed: ZERO_ADDRESS");
+
         oracleAdapter = IOracleAdapter(_oracleAdapterAddress);
         rateFeedID = _rateFeedID;
         borrowerOperations = IBorrowerOperations(_borrowerOperationsAddress);
         watchdogAddress = _watchdogAddress;
+
+        _transferOwnership(_initialOwner);
+    }
+
+    /**
+    * @notice Sets the watchdog address
+    * @param _watchdogAddress The address of the new watchdog contract
+    */
+    function setWatchdogAddress(address _watchdogAddress) external onlyOwner {
+        require(_watchdogAddress != address(0), "FXPriceFeed: ZERO_ADDRESS");
+
+        address oldWatchdogAddress = watchdogAddress;
+        watchdogAddress = _watchdogAddress;
+
+        emit WatchdogAddressUpdated(oldWatchdogAddress, _watchdogAddress);
     }
 
-    function fetchPrice() public returns (uint256, bool) {
+    /**
+    * @notice Fetches the price of the FX rate, if valid
+    * @dev If the contract is shutdown due to oracle failure, the last valid price is returned
+    * @return The price of the FX rate
+    */
+    function fetchPrice() public returns (uint256) {
```

Comment:
There's no access control here so anyone can call and update the lastValidPrice variable. I don't know if this is intentional but it could be used to front run price updates.